tmt/steps/provision/connect.py

```diff
@@ -31,12 +31,103 @@ class ConnectGuestData(tmt.steps.provision.GuestSshData):
         help='Username to use for all guest operations.'
         )
 
+    soft_reboot: Optional[ShellScript] = field(
+        default=None,
+        option='--soft-reboot',
+        metavar='COMMAND',
+        help="""
+             If specified, the command, executed on the runner, would be used
+             for soft reboot of the guest.
+             """,
+        normalize=tmt.utils.normalize_shell_script,
+        serialize=lambda value: str(value) if isinstance(value, ShellScript) else None,
+        unserialize=lambda serialized: None if serialized is None else ShellScript(serialized)
+        )
+    hard_reboot: Optional[ShellScript] = field(
+        default=None,
+        option='--hard-reboot',
+        help="""
+             If specified, the command, executed on the runner, would be used
+             for hard reboot of the guest.
+             """,
+        metavar='COMMAND',
+        normalize=tmt.utils.normalize_shell_script,
+        serialize=lambda value: str(value) if isinstance(value, ShellScript) else None,
+        unserialize=lambda serialized: None if serialized is None else ShellScript(serialized)
+        )
```

Comment:
Hmmm, I did not think about this. For me, it was a way to allow testing of a watchdog, against Beaker machines, from one's laptop. I did not think about anyone actually using the `connect` plugin in a plan I decided to run on my laptop :)

The idea with `--feeling-safe` sounds good, that could work out quite nicely.

Another possibility might be a `cli-only` flag for `field()`, allowing the field to be specified only on a command line, raising an exception if specified in an fmf node.

Reply:
> The idea with `--feeling-safe` sounds good, that could work out quite nicely.

Good! @lukaszachy, @thrix, any quick thoughts on this?

> Another possibility might be a `cli-only` flag for `field()`, allowing the field to be specified only on a command line, raising an exception if specified in an fmf node.

Sounds ok as well. But I'd probably prefer the `--feeling-safe` way because there are other use cases which will benefit from the functionality as well. I remember at least one:

```yaml
discover:
    how: shell
    generate: ./custom-script-generating-test-metadata.sh
```

So, let's rather make the first step in this direction?

Reply:
Yep, makes sense to add it first. I'll start right away, scratching our own itch...

Reply:
https://github.com/teemtee/tmt/pull/2513

Reply:
And now the PR is rebased on top of #2513, see `is_feeling_safe` in action at https://github.com/teemtee/tmt/pull/2512/files#diff-c053cd9bc1c960f6586dc3b0742c267cb2a9427e955d2ff23ff01ae9284be8a2R175

Reply:
Thanks, nice!